src/vs/workbench/contrib/positronNotebook/browser/positronNotebook.contribution.ts

```diff
@@ -76,24 +78,52 @@ class PositronNotebookContribution extends Disposable {
 						{ startDirty: false }
 					);
 
+					return { editor: editorInput, options };
+				},
+				createUntitledEditorInput: async ({ resource, options }) => {
+					// Determine notebook view type for untitled notebook
+					const viewType = await this.detectNotebookViewType(resource);
+
+					// Acquire a model reference so we can get the resolved resource for the untitled
+					// notebook. We *must* dispose this reference immediately after use to avoid
+					// leaking it which would otherwise keep the notebook document alive even after
+					// the editor has been closed (see https://github.com/positron/issues/XXXXX).
+					const ref = await this.notebookModelResolverService.resolve({ untitledResource: resource }, viewType);
+					const resolvedResource = ref.object.resource;
+					ref.dispose();
```

Comment:
okay here's another weird one - now when I click New File > Jupyter Notebook, it opens a VSC notebook even though my preference is Positron. When I save it, it switches to Positron. 🙃